### PR TITLE
Scheduler shutdown gracefully

### PR DESCRIPTION
**Pull request by peusebiu**
**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by peusebiu** [comment]
this will wait on all scheduler goroutines to finish when server is shutting down.

it will fix these types of test racing errors:

2023-10-20T08:06:10.2729247Z     testing.go:1206: TempDir RemoveAll cleanup: unlinkat /tmp/TestPeriodicallySignaturesErr2733177038: directory not empty

**Comment by rchincha** [comment]
Conflicts and rebase?

File: pkg/storage/scrub.go [hunk 1]
```diff
@@ -87,10 +88,6 @@ func CheckImageStoreBlobsIntegrity(ctx context.Context, imgStore storageTypes.Im
 func CheckRepo(ctx context.Context, imageName string, imgStore storageTypes.ImageStore) ([]ScrubImageResult, error) {
 	results := []ScrubImageResult{}
 
-	if ctx.Err() != nil {
```
[the remark this answers is not in the document]
Comment (peusebiu) [reply pkg/storage/scrub.go:90]: I think that would be too much, checking at the beginning of function and also in the for loop.

**Review by andaaron — APPROVED** [review]
lgtm

**Pull request merged**